js/index.js

```diff
@@ -0,0 +1,345 @@
+      setListItem($('.listNames ul').find("input:checked").val().concat('.Checked'), checked);
+    }
+  });
+  /*
+  * Function to delete list
+  */
+  function deleteList() {
+    //if the following is a list or list item
+    if ($(this).find('input').is(':radio')) {
+      //iterate through each
+      $(this).parent().each(function() {
+        //remove highlighting for each
+        $(this).toggleClass('highlight', false);
+      });
+    }
+    //if the following is a list or list item
+    if ($(this).prev().find('input').is(':radio'))
+      //delete list
+      deleteListHelper1.call(this);
+    else
+      //delete list item
+      deleteListHelper2.call(this);
+    //delete the list items
+    $('.listItems ul').empty();
+    var listItems = getListsItems($('.listNames ul').find("input:radio:checked").val());
+    var checked = getListsItems($('.listNames ul').find("input:radio:checked").val().concat('.Checked'));
+    //show list items and highlight checked
+    parseList(listItems, addItem);
+    parseChecked(checked);
+  }
+  /*
+  * Function to delete list from shopping list
+  */
+  function deleteListHelper1() {
+    //get the list
+    var shopLists = getLists();
+    //clear list items
+    $('.listItems ul').empty();
+    //remove selected list
+    shopLists.splice(shopLists.indexOf($(this).prev().find('input').val()), 1);
+    //remove list from local storage
+    //localStorage.removeItem($(this).prev().find('input').val());TODO redo based on this
+    setLists('Shopping Lists', shopLists);
+    //if the shopping list length is 2 due to '[]'
+    if (localStorage.getItem('Shopping Lists').length === 2) {
+      //remove the base item
+      localStorage.removeItem('Shopping Lists');
+      //set initial list
+      initialState();
+      //get list and list items
+      shopLists = getLists();
+      var listItems = getListsItems(shopLists[0]);
+      //display list and list items
+      parseList(shopLists, addListName);
+      parseList(listItems, addItem);
+    }
+  }
+  /*
+  * Function to delete item from a list
+  */
+  function deleteListHelper2() {
+    //get list items and what is checked
+    var listItems = getListsItems($('.listNames ul').find("input:radio:checked").val());
+    var checked = getListsItems($('.listNames ul').find("input:radio:checked").val().concat('.Checked'));
+    //remove item from checked list and then remove it from local storage
+    checked.splice(listItems.indexOf($(this).parent().find('input').val()), 1);
+    localStorage.removeItem($('.listNames ul').find("input:radio:checked").val().concat('.Checked'));
+    //remove item from list and remove from local storage
+    listItems.splice(listItems.indexOf($(this).prev().find('input').val()), 1);
+    setListItem($('.listNames ul').find("input:radio:checked").val(), listItems);
+    setLists($('.listNames ul').find("input:radio:checked").val().concat('.Checked'), checked);
+  }
+  /*
+  * Function to test if a storage type is available
```

Comment:
This issue will get resolved when project moves over to mvc architecture as project will be revamped.